### PR TITLE
32 feature settings menu

### PR DESCRIPTION
**Pull request by cmwall0605**

### CODE REVIEW

**Comment by RainbowPangolin** [comment]
The deployment seems broken. I can't start the timer or open any settings menu. Is this the final PR or is this meant to be a draft?

**Comment by cmwall0605** [comment]
The unit test issues are inherited from the issues of #65 so they'll be ignored for now

File: source/timer-page/timer.js [hunk 1]
```diff
@@ -81,37 +79,41 @@ function timerPageInit() {
 
   // set variable denote current timer mode
   localStorage.setItem('isPomo', 'false');
+
   // render current task name to timer page
   const id = JSON.parse(localStorage.getItem('currentTask'));
   allTasks = JSON.parse(localStorage.getItem('allTasks'));
-  if (allTasks && id) {
+  let flag = false;
```
Comment (RainbowPangolin) [inline source/timer-page/timer.js:86]: Variable needs a better name,

**Review by RainbowPangolin — COMMENTED** [review]
Some things should be renamed.

Also, avoid having refactoring changes with feature implementations in the same PR. I wasn't able to look through the whole PR because at some points I couldn't tell which changes were minor refactors or feature implementations.

**Comment by james-c-lars** [comment]
Definitely need to fix up the styling, there also appear to be some HTML issues that should be polished up ASAP: namely the labels should reference the input elements they're connected to using the for attribute, and the settings menu shouldn't use an h1 given that there's only meant to be a single h1 per page.

Additionally, the settings menu shouldn't be accessible during a timer session. Possibly it would make sense to be accessible during a break, but the option shouldn't exist while the timer is running for a work session. I'm also unsure whether it makes sense for the settings menu to be grouped with the other page options, when it only opens a modal rather than navigating to a new page. I do wonder if maybe having the settings exist solely on the timer page makes the most sense given that all our settings only interact with the timer and not the tasks list or the stats page.

With all that said I'm going to approve the changes given that we want our Web Components team to have access to the general changes ASAP.

**Review by james-c-lars — DISMISSED** [review]

**Review by james-c-lars — APPROVED** [review]

**Pull request merged**